content/en/BestPractices/SSTI-prevention.md

```diff
@@ -0,0 +1,179 @@
+
+#### Impact
+
+As a result of this usage of templates, both XSS vulnerabilities and SSTI vulnerabilities are created. The following code block is an example of an attacker’s input via the `name` parameter.
+
+```ts
+{{config.__class__.__init__.__globals__['os'].popen('ls').read()}}
+```
+
+
+### Bypassing the blacklisting
+
+The problem with blacklists is that they can often be circumvented, which is terrible.
+ 
+```ts
+@app.route("/page")
+blacklist = ["__class__",”request[request.”]
+def page():
+    name = request.values.get('name')
+    for bad_string in blacklist:
+        if  bad_string in name:
+            return "HACKING ATTEMPT {}".format(bad_string), 400
+        else:
+            output = Jinja2.from_string('Hello ' + name + '!').render()
+            return output    
+```
+
+#### Analysis
+
+It is possible to retrieve the value of a new `cobalt` GET parameter using the `request.args.param`. An attacker can bypass `__class__` using this. Since we have also blacklisted `request.args.param`, we will use jinja2 native function called `| attr().` 
+
+
+#### Impact
+
+An attacker can pass the following input to the server using the `name` parameter.
+
+```ts
+name={{request|attr('application')|attr('\x5f\x5fglobals\x5f\x5f')|attr('\x5f\x5fgetitem\x5f\x5f')('\x5f\x5fbuiltins\x5f\x5f')|attr('\x5f\x5fgetitem\x5f\x5f')('\x5f\x5fimport\x5f\x5f')('os')|attr('popen')('id')|attr('read')()}}
+```
+
+#### Prevention
+
+The variable `name` should be passed to the template context.
+
+
+```ts
+@app.route("/page")
+def page():
+    name = request.values.get('name')
+    output = Jinja2.from_string('Hello {{name}}!').render(name = name)
+    return output
+```
+
+
+## Best practices
+
+A template engine has advantages and facilitates modifications. For example, a design change can be made without touching the code flow and autonomously by the designers/integrators. The changes are independent of the code and logical processing. In the same way, developers can change logical elements and code without affecting data display or breaking the interface. It makes maintaining the project easier.
+For SSTI prevention, never allow users to modify or create templates. However, when business needs arise, there are solutions:
+
+#### Sanitization
+Before using a template, detecting and removing potentially malicious content is necessary. Data transmitted by users must therefore be efficiently analyzed. Various methods can be used to accomplish this (e.g., regex, whitelists of authorised expressions).
+There is, however, a drawback to this solution. In addition, configuration errors can put your environment at risk.
+
+#### Sandboxing
+The aim is to provide a closed environment where risky modules and features are disabled. For example, if user-supplied data is interpreted, it cannot provide access to other components or data.
+It may be challenging to create a sandbox environment, and misconfigurations may result in bypassing the environment.
```

Comment:
```suggestion
### Set Up a Sandbox Environment

You can turn off risky modules and features in a closed sandbox environment. This allows you to limit access to other application components.

It may be challenging to create a sandbox environment. Misconfigurations may result in bypassing the environment.
```